### ISSUE TITLE
Installation Error

### ISSUE THREAD

**Issue by VivekTodur** [Bioconductor/AnnotationFilter#18]
install_github("Bioconductor/AnnotationFilter")
Downloading GitHub repo Bioconductor/AnnotationFilter@master
from URL https://api.github.com/repos/Bioconductor/AnnotationFilter/zipball/master
Installing AnnotationFilter
'/usr/lib/R/bin/R' --no-site-file --no-environ --no-save --no-restore --quiet  \
  CMD INSTALL  \
  '/tmp/RtmpqRk882/devtools6fdd2b89cfc0/Bioconductor-AnnotationFilter-26a5c60'  \
  --library='/home/R/x86_64-pc-linux-gnu-library/3.4' --install-tests 

* installing *source* package ‘AnnotationFilter’ ...
** R
** tests
** preparing package for lazy loading
Warning: replacing previous import ‘stats::sd’ by ‘BiocGenerics::sd’ when loading ‘IRanges’
Warning: replacing previous import ‘stats::var’ by ‘BiocGenerics::var’ when loading ‘IRanges’
Warning: undefined slot classes in definition of "XRaw": elementMetadata(class "DataTableORNULL")
Warning: undefined slot classes in definition of "XInteger": elementMetadata(class "DataTableORNULL")
Warning: undefined slot classes in definition of "XDouble": elementMetadata(class "DataTableORNULL")
Error in (function (cl, name, valueClass)  : 
  assignment of an object of class “NULL” is not valid for @‘NAMES’ in an object of class “IRanges”; is(value, "characterORNULL") is not TRUE
Error : unable to load R code in package ‘AnnotationFilter’
ERROR: lazy loading failed for package ‘AnnotationFilter’
* removing ‘/home/R/x86_64-pc-linux-gnu-library/3.4/AnnotationFilter’
Installation failed: Command failed (1)

Could you please suggest solution for getting it installed...

**Comment by jorainer** [comment]
Could you please provide some more information on your R installation, such as the output of your `sessionInfo`. 
Note also that the default and supported way to install Bioconductor packages is by using the `BiocInstaller` package:
```r
source("https://bioconductor.org/biocLite.R")
biocLite("AnnotationFilter")
```
This ensures that all required packages are installed too.

**Comment by VivekTodur** [comment]
Thanks, for your quick response, Here is the output of the command,

> biocLite("AnnotationFilter")
BioC_mirror: https://bioconductor.org
Using Bioconductor 3.4 (BiocInstaller 1.24.0), R 3.4.0 (2017-04-21).
Installing package(s) ‘AnnotationFilter’
Warning message:
package ‘AnnotationFilter’ is not available (for R version 3.4.0) 
> sessionInfo()
R version 3.4.0 (2017-04-21)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 14.04.5 LTS

Matrix products: default
BLAS: /usr/lib/libblas/libblas.so.3.0
LAPACK: /usr/lib/lapack/liblapack.so.3.0

locale:
 [1] LC_CTYPE=en_IN       LC_NUMERIC=C         LC_TIME=en_IN       
 [4] LC_COLLATE=en_IN     LC_MONETARY=en_IN    LC_MESSAGES=en_IN   
 [7] LC_PAPER=en_IN       LC_NAME=C            LC_ADDRESS=C        
[10] LC_TELEPHONE=C       LC_MEASUREMENT=en_IN LC_IDENTIFICATION=C 

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] BiocInstaller_1.24.0

loaded via a namespace (and not attached):
[1] compiler_3.4.0 tools_3.4.0   

Thanks,

**Comment by jorainer** [comment]
It looks like you have an old Bioconductor version installed on a new R (most likely the system R got updated, but the Bioconductor libraries not). 
`AnnotationFilter` is available only since Bioconductor release 3.5 (and hence fails to install on your system because it depends on newer versions of installed packages).

In order to install and use `AnnotationFilter` you have first to upgrade your Bioconductor installation:
```r
source("https://bioconductor.org/biocLite.R")
biocLite()
```
The `BiocInstaller` version you have is outdated (version 1.24.0), you need 1.26.0
```r
> library(BiocInstaller)
Bioconductor version 3.5 (BiocInstaller 1.26.0), ?biocLite for help
```
